src/components/People.js

```diff
@@ -9,180 +9,164 @@ import SaveButton from "./SaveButton";
-                );
-            }.bind(this))
+class People extends React.Component {
+  state = {
+    data: [],
+    noSave: store.get("saved"),
+    justSaved: false,
+    updated_niceties: new Set()
+  }
+
+  saveAllComments = () => {
+      updated_niceties_spinlock = true;
+      let data_to_save = [];
+      const dateUpdated = new Date(Date.now());
+      const dateUpdatedStr = dateUpdated.toUTCString();
+      this.state.updated_niceties.forEach(function(e) {
+          const split_e = e.split(",");
+          let end_date;
+          if (split_e[1] === "null") {
+            end_date = null;
+          } else {
+            end_date = split_e[1];
+          }
+          const anonymous = store.get("anonymous-" + split_e[0], false);
+          const text = store.get("nicety-" + split_e[0], '');
+          const noRead = store.get("no_read-" + split_e[0], false);
+
+          data_to_save.push(
+              {
+                  target_id: parseInt(split_e[0], 10),
+                  end_date: end_date,
+                  anonymous: anonymous,
+                  text: text,
+                  no_read: noRead,
+                  date_updated: dateUpdatedStr
+              }
+          );
+      });
+      updated_niceties_spinlock = false;
+      $.ajax({
+          url: this.props.save_nicety_api,
+          data: {'niceties': JSON.stringify(data_to_save)},
+          dataType: 'json',
+          type: 'POST',
+          cache: false,
+          success: function() {
+              this.setState({noSave: true});
+              this.setState({justSaved: true});
+              store.set("saved", true);
+              this.state.updated_niceties.forEach(function(e) {
+                  const split_e = e.split(",");
+                  store.set("date_updated-" + split_e[0], dateUpdatedStr);
+              });
+              this.state.updated_niceties.clear();
+          }.bind(this),
+          error: function(xhr, status, err) {
+              console.log(err)
+          }
+      });
+  }
+
+  generateRows = (inputArray) => {
+      let dataList = [];
+      if (inputArray !== undefined) {
+          for (let i = 0; i < inputArray.length; i +=4) {
+              let row = [];
+              for (let j = 0; j < 4; j++) {
+                  if ((i + j) < inputArray.length) {
+                      row.push(inputArray[i + j]);
+                  }
+              }
+              dataList.push(row);
+          }
+      }
+      return dataList;
+  }
+
+  saveReady = () => {
+      this.setState({noSave: false});
+  }
+
+  alertTimer = () => {
+      setTimeout(function () {
+          this.setState({justSaved: false});
+      }.bind(this), 3000);
+  }
+
+  render() {
+      let leaving = this.generateRows(this.props.people.leaving);
+      let staying = this.generateRows(this.props.people.staying);
+      let faculty = this.generateRows(this.props.people.faculty);
+      const savePass = this.saveReady.bind(this);
```

Comment:
Likewise, `saveReady` is now an arrow function and does not need a `bind`, or to be in its own constant.